### PR TITLE
Feat/oauth registration

### PR DESCRIPTION
**Pull request by Zmicer-Yafremau**
#### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Performance optimization
- [ ] Refactoring
- [ ] Test Case
- [ ] Documentation update
- [ ] Other

#### 🔗 Related trello card link

[_Describe the source of requirement, like related issue link._](https://trello.com/c/2WTdOmOH/21-auth-complete-registration-component)

#### 💡 Background and solution

Pass email, first name and last name from OAuth to login and reg form

#### 📷 Screenshots

image

### CODE REVIEW

File: src/app/services/auth.service.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import { Subject } from 'rxjs';
+import { IUser } from '../models/user-model';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class AuthService {
+  public user = new Subject<IUser>();
+
+  public getUser(value: IUser) {
```
Comment (kubarenatioN) [inline src/app/services/auth.service.ts]: next is for setting value into subject. so getUser is incorrect naming for this case, use setUser instead

**Review by kubarenatioN — CHANGES_REQUESTED** [review]

File: src/app/services/auth.service.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { Injectable } from '@angular/core';
+import { Subject } from 'rxjs';
+import { IUser } from '../models/user-model';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class AuthService {
+  public user = new Subject<IUser>();
```
Comment (kubarenatioN) [inline src/app/services/auth.service.ts]: use BehaviorSubject here

File: src/app/services/auth.service.ts (hunk 1 above)
Comment (Zmicer-Yafremau) [reply src/app/services/auth.service.ts]: done

File: src/app/services/auth.service.ts (hunk 2 above)
Comment (Zmicer-Yafremau) [reply src/app/services/auth.service.ts]: done

**Review by ShArPman13 — APPROVED** [review]

**Review by kubarenatioN — APPROVED** [review]
Good job!

**Pull request merged**